packages/jellyfish-api-core/__tests__/category/net/getPeerInfo.test.ts

```diff
@@ -0,0 +1,106 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { PeerInfo } from 'packages/jellyfish-api-core/src/category/net'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import Dockerode from 'dockerode'
+
+createNetworkTests('Network without masternode', [new RegTestContainer(), new RegTestContainer()])
+createNetworkTests('Network on masternode', [new MasterNodeRegTestContainer(), new MasterNodeRegTestContainer()])
+
+function createNetworkTests (name: string, container: RegTestContainer[] | MasterNodeRegTestContainer[]): void {
+  describe(name, () => {
```

Comment:
Should not do this. 

Ideally, we don't want to introduce any abstraction at all to the test structure. Should take a different stance from writing code vs writing test. The test should be as straightforward, unoptimized, and as explicit as possible. 

You are trying to optimize the reading/reviewing experience of each test for the reviewer rather than the code itself. While introducing abstraction to optimize the code makes sense for non-test code, it creates a mental overhead for the reviewer.

Think of it as writing a story or narrative for your test, each scenario should be clear and obvious. 

Avoid abstraction, avoid introducing functions, avoid doing anything that is not part of the jest conventions. By doing so you are forcing the test conformity within the function. And you might lose IDE support, eslint support along the way with this pattern.

Reply:
Got it, will refactor to use proper conventions.